controller/authController.go

```diff
@@ -41,90 +36,103 @@ func Login(c *gin.Context) {
-			c.JSON(http.StatusNotFound, gin.H{"error": err.Error()})
-		} else if err.Error() == "invalid password"  {
-			c.JSON(http.StatusUnauthorized, gin.H{"error": err.Error()})
-		}else{
-			c.JSON(http.StatusInternalServerError, gin.H{"error": "Internal Server Error"})
-		}
+	if err !="" {
+		c.JSON(statusCode, gin.H{"error":err})
 		return
 	}
 
-	log.Printf("User %s signed up successfully", user.Username)
-	c.JSON(http.StatusOK, gin.H{"message": "Login successful"})
+	log.Printf("User %s logged up successfully", user.Username)
+	c.JSON(statusCode, gin.H{"message": "Login successful"})
 }
 
 
 var store = sessions.NewCookieStore([]byte("secret"))
 
 //Session Login
 func SessionLogin(c *gin.Context) {
-	var credentials model.User
-	if err := c.BindJSON(&credentials); err != nil {
+	var user model.User
+	if err := c.BindJSON(&user); err != nil {
 		log.Printf("Failed to bind JSON input for login: %v", err)
 		c.JSON(http.StatusBadRequest, gin.H{"error": "Cannot bind JSON input"})
 		return
 	}
 
-	err := credentials.Login(config.DB)
+	statusCode,err := user.Login()
 
-	if err != nil {
-		if err.Error() == "user not found"{
-			c.JSON(http.StatusNotFound, gin.H{"error": err.Error()})
-		} else if err.Error() == "invalid password"  {
-			c.JSON(http.StatusUnauthorized, gin.H{"error": err.Error()})
-		}else{
-			c.JSON(http.StatusInternalServerError, gin.H{"error": "Internal Server Error"})
-		}
+	if err !="" {
+		c.JSON(statusCode, gin.H{"error":err})
 		return
 	}
+		
 	// store - An instance of a session store created using the Gorilla Sessions package. 
 	//This store is responsible for managing session data and cookies.
-	//Get(c.Request, "mysession"): This method tries to retrieve an existing s
-	//ession by the name "mysession" from the user's browser cookies. 
+	//Get(c.Request, "mysession"): This method tries to retrieve an existing
+	//session by the name "mysession" from the user's browser cookies. 
 	//If a session with this name does not exist, a new session is created.
-	session, _ := store.Get(c.Request, "mysession")
-	session.Values["username"] = credentials.Username
+	session, _ := store.Get(c.Request, os.Getenv("SESSION_NAME"))
+	session.Values["username"] = user.Username
 
 	// session.Save (c.Request, c.Writer): This method writes the session cookie to the user's browser. 
 	//It takes the current HTTP request and response writer as arguments.
 	if err := session.Save(c.Request, c.Writer); err != nil {
 		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to save session"})
+		// log.Printf("Failed to save the session : %v", err)
 		return
 	}
-	log.Printf("User %s signed up successfully", credentials.Username)
+
+	log.Printf("User %s Logged In successfully", user.Username)
 	c.JSON(http.StatusOK, gin.H{"message": "Login successful"})
 }
 
 //Session Dashboard
 func Dashboard(c *gin.Context) {
-	session, _ := store.Get(c.Request, "mysession")
+	session, _ := store.Get(c.Request, os.Getenv("SESSION_NAME"))
```

Comment:
What's this session name field?

Reply:
Username is stored in the session so first we have to create a session with a name.